### PR TITLE
[SR GeekCamp] Support show materialized view in new Parser and new Analyzer

### LINKED ISSUE #4040: The new Parser and the new Analyzer support the related statements of materialized views [general]
### Requirement Description
The new Parser and the new Analyzer support the related statements of materialized views

### Mentor
@HangyuanLiu

### PR DESCRIPTION
**Pull request by Crystal-LiuJing**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4040

## Problem Summary(Required) ：

Support show materialized view in new Parser and new Analyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 8 / 17 (47.06%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -266,13 +266,20 @@ public ShowResultSet execute() throws AnalysisException {
...
-        String dbName = ((ShowMaterializedViewStmt) stmt).getDb();
+        ShowMaterializedViewStmt showMaterializedViewStmt = (ShowMaterializedViewStmt) stmt;
+        String dbName = showMaterializedViewStmt.getDb();
         List<List<String>> rowSets = Lists.newArrayList();
 
         Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
         if (db == null) {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
         }
+        if (!GlobalStateMgr.getCurrentState().getAuth().checkDbPriv(ctx, db.getFullName(), PrivPredicate.SHOW)) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:277]: better to check privilege in ShowStmtAnalyzer

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -178,6 +179,12 @@ public ParseNode visitShowTables(StarRocksParser.ShowTablesContext context) {
         }
     }
 
+    @Override
+    public ParseNode visitShowMaterializedView(StarRocksParser.ShowMaterializedViewContext context) {
+        String database = context.db != null ? context.db.getText() : null;
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:184]:
```suggestion
        String database = null;
        if (context.qualifiedName() != null) {
            database = getQualifiedName(context.qualifiedName()).toString();
        }
```

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:277]: move code to PrivilegeChecker

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 8 / 19 (42.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowMaterializedViewStmt.java|3|3|100.00%|

**Review by Astralidea — APPROVED** [review]

**Pull request merged**